project13/ViewController.swift

```diff
@@ -4,16 +4,191 @@
 //
 //  Created by Felipe Gil on 2021-07-09.
 //
-
+import CoreImage
 import UIKit
+//TODO
+#warning("Make the Change Filter button change its title to show the name of the currently filter")
+#warning("Try more than 1 slider with different properties")
 
-class ViewController: UIViewController {
-
+class ViewController: UIViewController, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    @IBOutlet weak var imageView: UIImageView!
+    @IBOutlet weak var intensity: UISlider!
+    
+    var currentImage: UIImage?
+    lazy var context: CIContext = CIContext()
+    var currentFilter: CIFilter!
+    
+    let loadingActivityIndicator: UIActivityIndicatorView = {
+        let indicator = UIActivityIndicatorView()
+        indicator.style = .large
+        indicator.color = .white
+        indicator.startAnimating()
+        indicator.autoresizingMask = [.flexibleLeftMargin, .flexibleRightMargin, .flexibleTopMargin, .flexibleBottomMargin]
+        return indicator
+    }()
+    let blurEffectView: UIVisualEffectView = {
+        let blurEffect = UIBlurEffect(style: .dark)
+        let blurEffectView = UIVisualEffectView(effect: blurEffect)
+        blurEffectView.alpha = 0.9
+        blurEffectView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        return blurEffectView
+    }()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        
+        let newTitle: String = "InstaFilter"
+        title = newTitle
+        
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(importPicture))
+
+        currentFilter = CIFilter(name: "CITwirlDistortion")
```

Comment:
As we have the `enum` as a replacement for the strings, this should read:

```suggestion
        currentFilter = CIFilter.make(.twirlDistortion)
```

Reply:
Now I see the reason for that - the `stringName` var of `CIFilterName` was broken. Please take a look at my comment for line 147.

Reply:
I've set the currentFilter this way in order to test, the problem on 147 was popping up when I started. 
On my ViewDidLoad, I'm setting the first filter. 
Fixed =D